File: simple.py
def outer_function(x):
    y = 10  # Local variable in the outer function
    
    def inner_function():
        nonlocal y  # Accessing the variable from the enclosing scope
        y += x
        print(f"Inner function: y = {y}")
    
    inner_function()
    print(f"Outer function: y = {y}")

# Dynamic typing example
var = 10
print(f"Type of var before: {type(var)}")
var = "Hello"
print(f"Type of var after: {type(var)}")

# Calling the function
outer_function(5)

File: simple.js
function outerFunction(x) {
    let y = 10; // Local variable

    function innerFunction() {
        y += x; // Accessing outer variable
        console.log(`Inner function: y = ${y}`);
    }

    innerFunction();
    console.log(`Outer function: y = ${y}`);
}

// Weak typing example
let variable = 10;
console.log(`Type before: ${typeof variable}`);
variable = "Hello";
console.log(`Type after: ${typeof variable}`);

// Calling the function
outerFunction(5);
